mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLFrameDecoder.kt

```diff
@@ -222,7 +222,8 @@ class MySQLFrameDecoder(val charset: Charset, private val connectionId: String)
 
         return if (this.totalColumns == this.processedColumns) {
             if (this.isPreparedStatementExecute) {
-                val row = slice.readBytes(slice.readableBytes())
+                val row = slice.retainedSlice()
+                slice.readerIndex(slice.readerIndex() + slice.readableBytes())
```

Comment:
one concern is the buffer is attached to BufferNotFullyConsumedException in BinaryRowDecoder, which may become unavailable during the exception dispatching.

The original code may leak iff buffer that returned from slice.readBytes must be explicitly release, but looks like API doc does not mention

Reply:
If not retain, the buf will be released while thie handler finished, and cannot be accessed in another handler.

Reply:
@andy-yx-chen  Replaced with `readRetainedSlice`.

Reply:
@jilen you are right, please help to check if there are other places that have the same issue here

Reply:
@oshai , we need you to take a look at this one, good finding actually